CONTRIBUTING.md

```diff
@@ -85,31 +100,48 @@ Binary output will be under each project's `bin\$(Configuration)\$(TargetFramewo
+
+To build, first-time run task `Restore` by pressing `Ctrl+P`, type `task Restore`, press `Enter`.
+After restoring packages you can execute build by pressing `Ctrl+Shift+B`.
+
+To run tests, highlight the name of the test with your cursor and press `Ctrl+P`, type `task DebugTest`, press `Enter`. If you don't want to build do `task DebugTest (No Build)` instead. Set your breakpoint in the test and press `F5` to start debugging by picking the dotnet process ID shown from `DebugTest`. Press `F5` once more to trigger the breakpoint.
 
 ### Creating a new Cmdlet and Unit Test
 
-1. Create a new class in a Module project and extend `PowerBICmdlet` from `Microsoft.PowerBI.Commands.Common` project and implement class.
-2. Add `[Cmdlet(CmdletVerb, CmdletName)]` attribute to class and provide `public const string` for `CmdletVerb` and `CmdletName`.
+1. Create a new class in a Module project.
+2. Either extend `PowerBIClientCmdlet` from `Microsoft.PowerBI.Common.Client` project or extend `PowerBICmdlet` from `Microsoft.PowerBI.Commands.Common` project if a client is not needed.
+3. Implement class and optionally add other interfaces from `Microsoft.PowerBI.Common.Abstractions` for common parameters.
+4. Add `[Cmdlet(CmdletVerb, CmdletName)]` attribute to class and provide `public const string` for `CmdletVerb` and `CmdletName`.
     * For `CmdletVerb`, pick from the System.Management.Automation.Verbs* classes.
-3. Optionally add `[OutputType(typeof(type))]` if your Cmdlet class if it writes an object to the output stream.
-4. Optionally add `[Alias("Get-Alias1", "Get-Alias2")]` to your Cmdlet class. 
-5. Add a test class to verify your cmdlet by adding (replacing with your new class name):
+5. Optionally add `[OutputType(typeof(type))]` if your Cmdlet class if it writes an object to the output stream.
+6. Optionally add `[Alias("Get-Alias1", "Get-Alias2")]` to your Cmdlet class. 
+7. Add a test class to verify your cmdlet by adding (replacing with your new class name):
 ```csharp
 using (var ps = System.Management.Automation.PowerShell.Create())
 {
+    ProfileTestUtilities.ConnectToPowerBI(ps); // If login is needed
     ps.AddCommand(new CmdletInfo($"{<class name>.CmdletVerb}-{<class name>.CmdletName}", typeof(<class name>))); // Optionally .AddParameter(), use intellisense to see options
     var result = ps.Invoke();
     // Add asserts to verify
+    TestUtilities.AssertNoCmdletErrors(ps);
 }
 ```
-6. If your test is interactive, add method attributes `[TestCategory("Interactive")]` and `[TestCategory("SkipWhenLiveUnitTesting")]` (last tells Live Unit Testing to skip it).
+8. If your test is interactive, add method attributes `[TestCategory("Interactive")]` and `[TestCategory("SkipWhenLiveUnitTesting")]` (last tells Live Unit Testing to skip it).
+9. Run test to verify cmdlet is being called correctly.
+10. Add parameters to cmdlet class and implement cmdlet logic in `ExecuteCmdlet()` method.
```

Comment:
Do you need a step about adding the cmdlet to CmdletsToExport?

Reply:
Yes I will add that